### DISCUSSION TITLE
Compatible version of python-engineio and python-socketio reports function argument mismatch

### DISCUSSION THREAD

**Discussion by meshashiranjan** [miguelgrinberg/python-socketio#1245] [Q&A] [answered]
**IMPORTANT**: If you have a question, or you are not sure if you have found a bug in this package, then you are in the wrong place. Hit back in your web browser, and then open a GitHub Discussion instead. Likewise, if you are unable to provide the information requested below, open a discussion to troubleshoot your issue.

Compatible version of python-engineio and python-socketio reports function argument mismatch

--- requirements.txt ----
Django
eventlet
python-engineio==**4.7.0**
python-socketio==**5.9.0**
pytz
six
uvicorn
requests
pyserial
-------

**To Reproduce**
1. Install package as shown in requirements.txt. With Django.
2. A socketio html page is having load of from https://cdnjs.cloudflare.com/ajax/libs/socket.io/**4.7.2**/socket.io.min.js
3. When the page loads
4. I get below error
5. 

**The error:**
GET /socket.io/?username=xxxxx&app_name=index_html&EIO=4&transport=polling&t=Ohbatxz HTTP/1.1" 200 278 0.001518
message handler error
Traceback (most recent call last):
  File "./venvpy/lib/python3.7/site-packages/engineio/server.py", line 644, in _trigger_event
    return self.handlers[event](*args)
  File "./venvpy/lib/python3.7/site-packages/socketio/server.py", line 783, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "./venvpy/lib/python3.7/site-packages/socketio/server.py", line 669, in _handle_connect
    'connect', namespace, sid, self.environ[eio_sid], data)
  File "./venvpy/lib/python3.7/site-packages/socketio/server.py", line 750, in _trigger_event
    return self.handlers[namespace][event](*args)
**TypeError: connect() takes 2 positional arguments but 3 were given**

**Expected behavior**
Socketio Connect should happen without any problem

**Comment by miguelgrinberg** [comment, marked as answer]
Please do not write issues on this project when you need help with an error in your own application. Thank you.

The error mentions a `connect()` function that takes 2 arguments, but it needs to take 3. My guess is that you do not have the `auth` argument on this function, which has authentication information passed by the client.
